### ISSUE TITLE
In-ellipsoids

### ISSUE THREAD

**Issue by tommy-waltmann** [glotzerlab/coxeter#162]
### Feature description

I would like to be able to get the in-ellipsoid of 2 and 3 dimensional shapes. The in-ellipsoid is analogous to the insphere in that its the ellipsoid with the largest volume that can fit inside the center of a shape, with the in-ellipsoid's center being shared with the actual shape.

### Proposed solution

the `inellipsoid` could just be a property accessible from different shape objects. As far as implementation goes, there is an open source matlab version here (https://www.mathworks.com/matlabcentral/fileexchange/24693-ellipsoid-fit), but a faster implementation would obviously be welcome.

### Additional context

To fix some issues I'm having with theory predicting especially elongated effective shapes, the insphere can be replaced with the inellipsoid to alter the shape kernel so the effective shapes predicted by theory can be correct. Having it available in coxeter would create a simple python API for my theory calculations.

**Comment by vyasr** [comment]
The concept seems fine, but please use a more descriptive name to clearly indicate the properties of the ellipse/ellipsoid. Recall that in coxeter the `insphere` is very specifically used for the case of a sphere tangent to all faces of a polyhedron, with `minimal_centered_bounded_sphere` probably being the sphere analog of what you really want. So probably `minimal_centered_bounded_ellips[e|oid]` is what you're going for.

**Comment by tommy-waltmann** [comment]
Do you mean `maximal_centered_bounded_sphere`, not `minimal_centered_bounded_sphere`? Either way, I think the proper name for this property should be `maximal_centered_bounded_ellipse`.

**Comment by vyasr** [comment]
Yup, that's what I mean, and your name sounds right to me.

**Comment by tommy-waltmann** [comment]
I've looked further into what would be needed to implement this feature. The implementation I referenced in the previous post is an implementation for fitting an ellipsoid, not finding the maximal inscribed ellipse, so that can't be used to implement this feature.

I've looked further into what would be needed, and the solution to this problem involves solving a minimization problem of one variable subject to a few constraints. We would likely need another package (like https://www.cvxpy.org/) to solve the minimization problem.

For my own use case, I can get by without the true maximal inscribed ellipse, and I'm not very motivated to learn another package to implement a feature that isn't a must-have.

I'm going to close this issue for now. It's possible I'll need the true maximal ellipse at some point, at which point I may re-open this.
